test/jdk/java/nio/channels/Channels/TransferTo.java

```diff
@@ -59,32 +64,34 @@
 
     private static final Random RND = RandomFactory.getRandom();
 
-    public static void main(String[] args) throws Exception {
-        test(fileChannelInput(), fileChannelOutput());
-        test(readableByteChannelInput(), defaultOutput());
+    @DataProvider
+    public static Object[][] streamCombinations() throws Exception {
+        return new Object[][] {
+            { fileChannelInput(), fileChannelOutput() },
+            { readableByteChannelInput(), defaultOutput() }
+        };
     }
 
-    private static void test(InputStreamProvider inputStreamProvider, OutputStreamProvider outputStreamProvider)
-            throws Exception {
-        testNullPointerException(inputStreamProvider);
-        testStreamContents(inputStreamProvider, outputStreamProvider);
+    @Test(dataProvider = "streamCombinations", expectedExceptions = NullPointerException.class)
```

Comment:
We have been moving away from using the  expectedExceptions annotation element and using the assertThrows method instead, example:

assertThrows(NullPointerException.class, () -> inputStreamProvider.input().transferTo(null));

Reply:
Done in https://github.com/openjdk/jdk/pull/5421/commits/b37ad6b39ffc78af7ca8c487125b7e4e0c1ff141.